protocol/csv/csvforwarder.go

```diff
@@ -57,6 +58,22 @@ func (connector *FilenameForwarder) AddDatapoints(ctx context.Context, points []
 	return nil
 }
 
+// AddEvents writes the events to a file
+func (connector *FilenameForwarder) AddEvents(ctx context.Context, events []*event.Event) error {
```

Comment:
Can you use a lock to prevent AddEvents and AddDatapoints from colliding since they can be happening in different goroutines